nearcore/src/config.rs

```diff
@@ -1257,21 +1257,29 @@ impl From<NodeKeyFile> for KeyFile {
     }
 }
 
-pub fn load_config(dir: &Path, genesis_validation: GenesisValidationMode) -> NearConfig {
-    let config = Config::from_file(&dir.join(CONFIG_FILENAME)).unwrap();
+pub fn load_config(
+    dir: &Path,
+    genesis_validation: GenesisValidationMode,
+) -> anyhow::Result<NearConfig> {
+    let config = Config::from_file(&dir.join(CONFIG_FILENAME))?;
     let genesis_file = dir.join(&config.genesis_file);
-    let validator_signer = if dir.join(&config.validator_key_file).exists() {
+    let validator_file = dir.join(&config.validator_key_file);
+    let validator_signer = if validator_file.exists() {
         let signer =
-            Arc::new(InMemoryValidatorSigner::from_file(&dir.join(&config.validator_key_file)))
-                as Arc<dyn ValidatorSigner>;
+            Arc::new(InMemoryValidatorSigner::from_file(&validator_file).with_context(|| {
+                format!("Failed initializing validator signer from {}", validator_file.display())
+            })?) as Arc<dyn ValidatorSigner>;
```

Comment:
This expression is a bit too busy now, with both error handling and `as` casing. I'd re-shuffled it a bit

```reust
let signer = InnMemoryValidatorSigner::from_file(&validator_file).with_context(|| {
    format!("Failed initializing validator signer from {}", validator_file.display())
})?;
Some(Arc::new(signer) as Arc<dyn ValidatorSigner>)
```